### PR TITLE
(PA-159) Ruby bits for huaweios-6-ppc

### PR DESCRIPTION
**Pull request by ScottGarman**
All the tasty morsels required to cross-compile ruby and associated gems.

### CODE REVIEW

File: configs/components/ruby.rb [hunk 1]
```diff
@@ -143,6 +156,7 @@
     # This tells the ruby setup that it can use the default system gcc rather
     # than our own.
     target_dir = File.join(settings[:libdir], "ruby", "2.1.0", rbconfig_info[settings[:platform_triple]][:target_double])
+    sed = "sed"
     sed = "gsed" if platform.is_solaris?
     sed = "/opt/freeware/bin/sed" if platform.is_aix?
```
Comment (melissa) [inline configs/components/ruby.rb]: this makes me sad

File: configs/components/ruby-augeas.rb [hunk 2]
```diff
@@ -24,6 +24,10 @@
       pkg.environment "RUBY" => settings[:host_ruby]
     end
     ruby = "#{settings[:host_ruby]} -r#{settings[:datadir]}/doc/rbconfig.rb"
+  elsif platform.is_huaweios?
+    pkg.environment "RUBY" => settings[:host_ruby]
+    ruby = "#{settings[:host_ruby]} -r#{settings[:datadir]}/doc/rbconfig.rb"
+    pkg.environment "LDFLAGS" => settings[:ldflags]
```
Comment (mckern) [inline configs/components/ruby-augeas.rb]: We don't set LDFLAGS like this by default?

File: configs/components/ruby.rb [hunk 3]
```diff
@@ -143,6 +156,7 @@
     # This tells the ruby setup that it can use the default system gcc rather
     # than our own.
     target_dir = File.join(settings[:libdir], "ruby", "2.1.0", rbconfig_info[settings[:platform_triple]][:target_double])
+    sed = "sed"
     sed = "gsed" if platform.is_solaris?
```
Comment (mckern) [inline configs/components/ruby.rb]: This is sort of meh but I know you didn't make the mess. Can we clarify intent a little with something like:

```ruby
sed = platform.is_solaris? "gsed" : "sed"
```

File: configs/components/ruby-augeas.rb (hunk 2 above)
Comment (shrug) [inline configs/components/ruby-augeas.rb]: We *should* but I'm not sure we do, I've sen this declared like this elsewhere too.

Comment (shrug) [inline configs/components/ruby-augeas.rb]: Poking through vanagon, we do not set a default for ldflags anywhere. feature idea!

File: configs/components/ruby.rb (hunk 1 above)
Comment (mckern) [inline configs/components/ruby.rb]: Oh, I didn't even notice the third option. Good glob, what a mess.

**Comment by mckern** [comment]
Some nitpicks, but no blockers. I wonder how long it'd take to abstract out cross compiling before the next platform.

File: configs/components/ruby.rb (hunk 3 above)
Comment (shrug) [inline configs/components/ruby.rb]: case statement?

Comment (mckern) [inline configs/components/ruby.rb]: You'd think so, but try to construct it and see what happens.

Comment (shrug) [inline configs/components/ruby.rb]: Nope, I'm dumb, that wouldn't work.

**Comment by shrug** [comment]
:+1:

**Pull request merged**